### PR TITLE
Documented about how to change default password in README.md

### PR DESCRIPTION
**Pull request by biswajit-9776**
# Pull Request Template for Kubeflow manifests Issues

#2777 
## ✅ Contributor checklist
  - Make sure you have tested with kustomize. See [Installation Prerequisites](https://github.com/kubeflow/manifests#prerequisites)
  - All the commits have been _signed-off_  (To pass the `DCO` check)
  - Submit the [Contributor License Agreements](https://cla.developers.google.com/clas) (To pass the `cla/google` check)

---     
 
> You can join the CNCF Slack and access our meetings at the [Kubeflow Community](https://www.kubeflow.org/docs/about/community/) website. Our channel on the CNCF Slack is here [**#kubeflow-platform**](https://app.slack.com/client/T08PSQ7BQ/C073W572LA2).

### CODE REVIEW

**Comment by biswajit-9776** [comment]
Hi, I have tested the _before creating cluster_ locally and it works fine. @diegolovison do let me know if you wish me to add anything else.

File: README.md [hunk 1]
```diff
@@ -465,25 +465,46 @@ If you absolutely need to expose Kubeflow over HTTP, you can disable the `Secure
...
+    kubectl delete secret dex-passwords -n auth
+    ```
+
+3. Create secret dex-passwords with new hash using the following command:
+
+    ```sh
+    kubectl create secret generic dex-passwords --from-literal=DEX_USER_PASSWORD='<place the generated hash here' -n auth
+    ```
+
+4. Restart the _dex_ pod in auth namespace using the following command:
```
Comment (diegolovison) [inline README.md:500]: Maybe replace `Restart` with `Recreate` ?

File: README.md [hunk 2]
```diff
@@ -465,25 +465,46 @@ If you absolutely need to expose Kubeflow over HTTP, you can disable the `Secure
...
+1. Delete the existing secret _dex-passwords_ in auth namespace using the following command:
+
+    ```sh
+    kubectl delete secret dex-passwords -n auth
+    ```
+
+3. Create secret dex-passwords with new hash using the following command:
+
+    ```sh
+    kubectl create secret generic dex-passwords --from-literal=DEX_USER_PASSWORD='<place the generated hash here' -n auth
```
Comment (diegolovison) [inline README.md:497]: `<place the generated hash here` with `REPLACTE_WITH_HASH` this brings the user more attention to that.

File: README.md [hunk 3]
```diff
@@ -465,25 +465,46 @@ If you absolutely need to expose Kubeflow over HTTP, you can disable the `Secure
 
 ### Change default user password
 
-For security reasons, we don't want to use the default password for the default Kubeflow user when installing in security-sensitive environments. Instead, you should define your own password before deploying. To define a password for the default user:
+For security reasons, we don't want to use the default password for the default Kubeflow user when installing in security-sensitive environments. Instead, you should define your own password and apply it either **before creating the cluster** or **after creating the cluster**. 
 
-1. Pick a password for the default user, with email `user@example.com`, and hash it using `bcrypt`:
-
-TODO this changed slightly in https://github.com/kubeflow/manifests/pull/2669 and https://github.com/kubeflow/manifests/pull/2229
+Pick a password for the default user, with email `user@example.com`, and hash it using `bcrypt`:
```
Comment (diegolovison) [inline README.md:470]: Give an example how to do it with `bcrypt`

File: README.md [hunk 4]
```diff
@@ -465,25 +465,46 @@ If you absolutely need to expose Kubeflow over HTTP, you can disable the `Secure
...
+
+1. Edit `common/dex/base/dex-passwords.yaml` and fill the relevant field with the hash of the password you chose:
 
     ```yaml
     ...
-      staticPasswords:
-      - email: user@example.com
-        hash: <enter the generated hash here>
+      stringData:
+        DEX_USER_PASSWORD: <replace the generated hash here>
```
Comment (diegolovison) [inline README.md:483]: `<place the generated hash here` with `REPLACTE_WITH_HASH` this brings the user more attention to that.

**Review by diegolovison — CHANGES_REQUESTED** [review]

**Comment by diegolovison** [comment]
If you have time and wish please create a new PR how to change the username

File: README.md (hunk 1 above)
Comment (biswajit-9776) [reply README.md]: Would you recommend if we use the following command instead of the delete command?
`kubectl rollout restart deployment dex -n auth`

Comment (diegolovison) [reply README.md]: cc @juliusvonkohout

**Comment by juliusvonkohout** [comment]
Lgtmed by documentation lead @diegolovison so I 

/approve

Since it should improve the status quo.

**Pull request merged**